Feature/parking debugger

Gursaanj opened the pull request:
Created  a **parking spot debugger tool**, that lets us choose which parking spot we want to set as the current destination, a maybe useful tool to test out new parking spots. 

You can't see it clearly in the video but here are the details

- The window only works in Play mode, as soon as you go back to editor mode, the window closes
- The window works between scenes, meaning you can carry the window from one map to another in one play session and it still work 
- When you select a parking Spot, it highlights the position of the parking spot in the Scene view, so that you can see where the parking spot is before you decide to change it 

![parking spot debugger](https://user-images.githubusercontent.com/23490604/81522488-9efe4080-92ff-11ea-9be4-356551c32808.gif)

Gursaanj commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create Parking Spot Debugging Tool](https://trello.com/c/7NxWs3cc/516-create-parking-spot-debugging-tool)

m4rtint commented on Assets/Editor/PartyParkadeParkingDebugger.cs:
```diff
@@ -0,0 +1,110 @@
+        m_window.Show();
+
+        // To avoid complications with using the window between edit and PlayMode
+        EditorApplication.playModeStateChanged += delegate { m_window.Close(); };
```
If you keep opening the window, wouldn't this keep attaching to the delegate, creating memory leak?

Gursaanj replied:
Good point, the example I saw for this never unsubscribed so I thought it was okay but, no harm unsubscribing on destroy

m4rtint commented on Assets/Editor/PartyParkadeParkingDebugger.cs:
```diff
@@ -0,0 +1,110 @@
+            m_currentParkingSpotIndex = GUILayout.SelectionGrid(m_currentParkingSpotIndex, parkingSpotNames, 3, GUILayout.Height(350f));
+            m_currentParkingSpot = m_parkingSpots[m_currentParkingSpotIndex];
+            Selection.activeGameObject = m_currentParkingSpot;
+            //SceneView.FrameLastActiveSceneView();
```
Any news on this problem??

Gursaanj replied:
Not yet, Like it works but it sometimes crashes the editor, so im keeping it here for now and maybe when I figure it Ill make it work!!

m4rtint commented on Assets/_PartyParkade/Script/Managers/ParkingManager.cs:
```diff
@@ -30,6 +30,22 @@ public class ParkingManager : MonoBehaviour
     /// <value>The current target.</value>
     private GameObject m_currentTarget = null;
 
+    public GameObject[] Targets
```
Targets => m_targets is less lines of code

m4rtint commented on Assets/Editor/PartyParkadeParkingDebugger.cs:
```diff
@@ -0,0 +1,124 @@
+
+            if (m_parkingSpots != null)
+            {
+                m_currentParkingSpot = m_parkingManager.CurrentTarget;
```
not sure if this is a logic mix up, but why use m_parkingManager here, when you already checked the null value in the if statement above

m4rtint replied:
Should also test what happens if there are no parking spots on the map at all.

m4rtint commented:
Do you know why Assets/SoundManagerPro/Scripts/Extensions/DFGUI_SMPRO_INTEGRATION.unitypackage.meta
is added?

m4rtint commented:
![image](https://user-images.githubusercontent.com/5779541/81627917-17bed480-93b4-11ea-9d2b-a726b3eaba17.png)
If there are too many parking slots, UI is messed up, maybe look into scroll view

m4rtint commented:
[Party-Parkade - ResidentialMap - PC, Mac & Linux Standalone - Unity 2019.3.0f6 Personal [PREVIEW PACKAGES IN USE] _DX11_ 2020-05-11 18-22-56.mp4.zip](https://github.com/PerigonGames/Party-Parkade/files/4612873/Party-Parkade.-.ResidentialMap.-.PC.Mac.Linux.Standalone.-.Unity.2019.3.0f6.Personal.PREVIEW.PACKAGES.IN.USE._DX11_.2020-05-11.18-22-56.mp4.zip)

https://1drv.ms/u/s!ArQDn0JFxY433ydcU0heNYe-mEiA?e=5PJsLB

- Particle Effects don't move with the spot 
- Residential Map has one broken spot

Gursaanj commented:
So this has been updated to allow for a dynamic scroll view of parking spots, so if there are 3 or 300 parking spots in a map, the scroll view should be able to handle everything cleanly

Fixed the sceneview bug, so that when you select a parking spot, it will move the scenview camera accordingly  to better visualize what parking spot you are looking at

Fixed the bug  where, the panel will update automatically when you park the car in a spot and a new spot has been chosen
